src/components/profile/Meetings.tsx

```diff
@@ -345,8 +350,48 @@ const Meetings: React.FC<{ currentAccount: Account }> = ({
   const { CancelDialog, openCancelDialog } = useCancelDialog()
   const fillMeeting = async () => {
     try {
-      const meeting = await getMeeting(slotId as string)
-      const decodedMeeting = await decodeMeeting(meeting, currentAccount!)
+      let meeting: ExtendedDBSlot
+      let decodedMeeting: MeetingDecrypted | null = null
+
+      if (conferenceId) {
+        // For dashboard, getSlotByMeetingId should return the account slot for the current user
+        const slot = await getSlotByMeetingId(conferenceId as string)
+        if (!slot) {
+          toast({
+            title: 'Meeting not found',
+            status: 'error',
+            description: 'The meeting you are trying to access was not found.',
+            duration: 5000,
+            isClosable: true,
+          })
+          return
+        }
+        // For dashboard meetings, we expect account slots
+        // Single calendar events: both host and guest share the same meeting_id (conferenceId)
+        // but each has their own slot. getSlotByMeetingId should return the slot for the current account
+        if (slot.user_type === 'account') {
+          meeting = slot as ExtendedDBSlot
+          decodedMeeting = await decodeMeeting(meeting, currentAccount!)
+        } else {
```

Comment:
The meeting can, I've already implemented this, make reference to the ScheduleMain component